src/baseApi.ts

```diff
@@ -4,16 +4,31 @@ import { ApiResponse } from './utils/response/apiResponse';
 const DEFAULT_API_VERSION = 3;
 const DEFAULT_HOST = 'go.timetac.com';
 
+export type Tokens = {
+  accessToken: string;
+  refreshToken: string;
+};
+
+type onTokenRefreshedCallback = (tokens: Tokens) => void;
+
+export type ApiState = {
+  refreshingToken: false | Promise<AxiosResponse<any>>;
```

Comment:
I like the types here :+1: 
But for `refreshingToken` I would expect just a boolean, so I would consider:
```suggestion
  refreshTokenPromise?: Promise<AxiosResponse<TokenResponse>>;
```

Reply:
As discussed promise offers better usage.